addons/account/models/account_payment.py

```diff
@@ -637,6 +637,8 @@ def _check_payment_method_line_id(self):
         for pay in self:
             if not pay.payment_method_line_id:
                 raise ValidationError(_("Please define a payment method line on your payment."))
+            elif pay.payment_method_line_id not in pay.available_payment_method_line_ids:
```

Comment:
Hello :)

This is what I originally did, it would require less tests to be changed, but some will still need to be.
Like the tests in addons/account/tests/test_reconciliation.py (they use a different journal but still use the default payment method of the setup class)